src/Mod/Draft/draftobjects/patharray.py

```diff
@@ -129,34 +159,34 @@ def getWireFromSubs(self,obj):
                 sl.append(e)
         return Part.Wire(sl)
 
-    def pathArray(self,shape,pathwire,count,xlate,align):
-        '''Distribute shapes along a path.'''
-        import Part
-
-        placements = calculatePlacementsOnPath(
-            shape.Placement.Rotation, pathwire, count, xlate, align)
-
-        base = []
-
-        for placement in placements:
-            ns = shape.copy()
-            ns.Placement = placement
-
-            base.append(ns)
-
-        return (Part.makeCompound(base))
+    def onDocumentRestored(self, obj):
+        self.setProperties(obj)
 
+        self.migrate_attributes(obj)
+        if self.use_link:
+            self.linkSetup(obj)
+        else:
+            obj.setPropertyStatus('Shape','-Transient')
+        if obj.Shape.isNull():
+            if getattr(obj,'PlacementList',None):
+                self.buildShape(obj,obj.Placement,obj.PlacementList)
+            else:
+                self.execute(obj)
 
 _PathArray = PathArray
 
-
-def calculatePlacementsOnPath(shapeRotation, pathwire, count, xlate, align):
+def calculatePlacementsOnPath(shapeRotation, pathwire, count, xlate, align,
+        mode = 'Original', forceNormal=False, normalOverride=None):
     """Calculates the placements of a shape along a given path so that each copy will be distributed evenly"""
     import Part
     import DraftGeomUtils
-
     closedpath = DraftGeomUtils.isReallyClosed(pathwire)
+
     normal = DraftGeomUtils.getNormal(pathwire)
+    if (forceNormal):
+        if not normalOverride is None:
```

Comment:
The Python way is
```py
if normalOverride is not None:
```

Or better
```py
if normalOverride:
```

This is two ands
```py
if forceNormal and normalOverride:
```

Reply:
Done.